### PR TITLE
onboarding navigation issues (back button, etc) #115

### PR DESCRIPTION
**Pull request by situatedbit**
This is a substantial PR. I'm addressing several bugs in #115, many of which I discovered during debugging.

@IagoSRL _Please Note_: there is a database update script to change the onboarding status of users on the live database.

Let me know if you want to review this together.

## Refactored OnboardingProgress and related code
- Removed unused groups subsets of onboarding steps (includes database changes)
- Added more semantic methods
- Onboarding API methods update user's onboarding progress only if user is already on the furthest activity in the process
- moved window referrer logic out of onboarding and into appInit
- removed commented code

## learnMoreProfessional
- we were losing the default jobTitleID on sign ups from learnMoreProfessional

## HashbangHistory
- uses native replaceState method (because browser support now permits it)
- fixed bug which prevented browser forward button from going all the way forward

## servicesOverview
- fixed typo in GroupRemoteModel which caused jobTitleID to be lost if navigated away from servicesOverview and then back again

## Sub-Dialogs
- Instead of using app-wide successSave method, use shell.goBack() to pull the sub-dialog out of the history list during onboarding

### CODE REVIEW

**Comment by situatedbit** [comment]
I just added a few changes to the copy on onboarding pages to let users know that they can always add more items/details after onboarding.

File: app/source/js/activities/licensesCertificationsForm.js [hunk 1]
```diff
@@ -209,7 +209,7 @@ function ViewModel(app) {
             app.model.jobTitleLicenses.clearCache();
             app.model.userLicensesCertifications.clearCache();
             // Go out
-            app.successSave();
+            app.shell.goBack();
```
Comment (IagoSRL) [inline app/source/js/activities/licensesCertificationsForm.js]: Same problem as commented at *addressEditor.js*, and repeated on line 240 of this file.

File: app/source/js/activities/serviceProfessionalServiceEditor.js [hunk 2]
```diff
@@ -36,8 +36,7 @@ var A = Activity.extend(function ServiceProfessionalServiceEditorActivity() {
             this.app.shell.goBack(this.requestData);
         }
         else {
-            // Just execute the standard save process
-            this.app.successSave();
+            this.app.shell.goBack();
```
Comment (IagoSRL) [inline app/source/js/activities/serviceProfessionalServiceEditor.js]: Same as commented for *addressEditor.js*

**Comment by situatedbit** [comment]
Thanks! Sounds good. I made the changes you requested. 

I haven't actually seen the save message, but I also didn't look carefully into why it isn't showing up. If we get to #264 I think ``showNavBarNotification()`` will be useful.

**Comment by IagoSRL** [comment]
Thanks and good job @situatedbit!

I keep the branch since from your comments it seems there is still more work on #115. If not the case, let me know and I will archive it and close.

**Pull request merged**